packages/block-editor/src/components/contrast-checker/contrast-checker-message.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { speak } from '@wordpress/a11y';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+
+export default function ContrastCheckerMessage( {
+	colordBackgroundColor,
+	colordTextColor,
+	colordLinkColor,
+	backgroundColor,
+	textColor,
+	linkColor,
+	shouldShowTransparencyWarning,
+} ) {
+	let msg = '';
+	if ( shouldShowTransparencyWarning ) {
+		msg = __( 'Transparent text may be hard for people to read.' );
+	} else {
+		const backgroundColorBrightness = colordBackgroundColor.brightness();
```

Comment:
In another iteration we could maybe make this a cleaner function and remove the message logic, passing the message only.